sdk/core/Azure.Core/src/Pipeline/Internal/ReadClientRequestIdPolicy.cs

```diff
@@ -1,10 +1,15 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Threading;
+
 namespace Azure.Core.Pipeline
 {
     internal class ReadClientRequestIdPolicy : HttpPipelineSynchronousPolicy
```

Comment:
It's worth thinking about whether or not we could generalize this a bit in the future.  I'd like to implement Storage's server-side timeouts (and other rarely used headers) the same way.  If it were possible to have a shared source RequestScope/RequestScopePolicy that let me override a private virtual to attach my own header, it would be enough to avoid me doing my own thing.

Reply:
Yes, I was thinking about Storage's server-side timeouts while doing this change.

Seems like having a property on client options for server timeout would be a more user-friendly solution.

Reply:
[Server-side timeouts vary by operation](https://docs.microsoft.com/en-us/rest/api/storageservices/setting-timeouts-for-blob-service-operations#exceptions-to-default-timeout-interval) and I'd prefer not to create a new pipeline for every operation.  It’s also advanced enough that I wouldn’t want people confusing it with client-side timeouts that most of them should be using instead.

Reply:
Sure, but how often would you want to change them on per operation basis. Especially considering 30 sec is the max value.

Reply:
When I say client-side timeout, I'm talking about using `CancellationTokenSource` which is configurable per call.  And I think server-side timeouts are all advanced and/or niche scenarios.  We could ping Jeff for more examples of customers needing to customize it.